### PR TITLE
Robustify the log message anonymisation: 

### PR DESCRIPTION
**Pull request by jerome-benoit**
+ Fix the recursion ending conditions and call;
+ Log missed message anonymisation; 
+ Anonymize also message string (probably not needed but in case of mistake, it's better do not disclose sensitive information).

### CODE REVIEW

File: src/migration/tasks/AddNotificationsFlagsToUsersTask.ts [hunk 1]
```diff
@@ -49,24 +49,24 @@ export default class AddNotificationsFlagsToUsersTask extends MigrationTask {
...
-        sendChargingStationRegistered: user.notifications.sendChargingStationRegistered ? user.notifications.sendChargingStationRegistered : false,
-        sendOcpiPatchStatusError: user.notifications.sendOcpiPatchStatusError ? user.notifications.sendOcpiPatchStatusError : false,
-        sendSmtpAuthError: user.notifications.sendSmtpAuthError ? user.notifications.sendSmtpAuthError : false,
-        sendOfflineChargingStations: user.notifications.sendOfflineChargingStations ? user.notifications.sendOfflineChargingStations : false,
-        sendBillingSynchronizationFailed: user.notifications.sendBillingSynchronizationFailed ? user.notifications.sendBillingSynchronizationFailed : false,
-        sendCarCatalogSynchronizationFailed: user.notifications.sendCarCatalogSynchronizationFailed ? user.notifications.sendCarCatalogSynchronizationFailed : false,
-        sendEndUserErrorNotification: user.notifications.sendEndUserErrorNotification ? user.notifications.sendEndUserErrorNotification : false,
+        sendSessionStarted: user.notifications?.sendSessionStarted ? user.notifications.sendSessionStarted : false,
```
Comment (LucasBrazi06) [inline src/migration/tasks/AddNotificationsFlagsToUsersTask.ts:52]: ? works only for optional parameter and Angular template

File: src/utils/Logging.ts [hunk 2]
```diff
@@ -658,36 +658,42 @@ export default class Logging {
...
-                key === 'captcha') {
-              // Anonymize
-              message[key] = Constants.ANONYMIZED_VALUE;
-            }
-          }
-        }
-      }
+    } else if (typeof message === 'string') {
+      // Anonymize
+      message.replace(/((repeat|)[pP]assword|captcha)(\s)(=|:)(\s)(.*)/g, '$1$3$4$5' + Constants.ANONYMIZED_VALUE);
```
Comment (LucasBrazi06) [inline src/utils/Logging.ts:666]: Not understandable by a human being.

File: src/migration/tasks/AddNotificationsFlagsToUsersTask.ts (hunk 1 above)
Comment (jerome-benoit) [reply src/migration/tasks/AddNotificationsFlagsToUsersTask.ts:52]: It's a typescript syntax extension enabled by default.

> PR again melted with import sorting -> difficult to read, I want only few files to review in this PR

Disable the buggy code extensions that do imports sorting like I've already told you and that will never happen again. Every developer must disable it.

File: src/utils/Logging.ts (hunk 2 above)
Comment (jerome-benoit) [reply src/utils/Logging.ts:666]: The regex is simple, more simple than the ones you can read in Constants.ts.
The replace https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/String/replace documentation is clear enough to explain what is done.

File: src/migration/tasks/AddNotificationsFlagsToUsersTask.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/migration/tasks/AddNotificationsFlagsToUsersTask.ts:52]: > It's a typescript syntax extension enabled by default.
Weird, it didn't work at the time I tested this in ts files but only in template!

Comment (LucasBrazi06) [reply src/migration/tasks/AddNotificationsFlagsToUsersTask.ts:52]: > Disable the buggy code extensions that do imports sorting like I've already told you and that will never happen again. Every developer must disable it.

I disabled sort-imports already, is it the right one?

I use this one:
image

Which is related to the Auto Import here:
image

**Comment by LucasBrazi06** [comment]
Ureadable PR

File: src/migration/tasks/AddNotificationsFlagsToUsersTask.ts (hunk 1 above)
Comment (jerome-benoit) [reply src/migration/tasks/AddNotificationsFlagsToUsersTask.ts:52]: No, it's a code extension named sort-imports, not a code settings related to automatic importing.

**Comment by jerome-benoit** [comment]
> Unreadable PR

I'm not going to spend more time on fixing the logic, the level of security and the performance of the recursion used to do log message anonymisation during my vacations.

A correct, secure and optimised implementation is available here in case you're interested: https://github.com/jerome-benoit/ev-server/blob/anonymisation-fixes/src/utils/Logging.ts

**Pull request closed without merging**